File: frontend/src/types/ap-insights.ts
export type APInsightSeverity = 'critical' | 'warning' | 'info'

export type APInsightActionType =
  | 'create_policy'
  | 'open_workbench'
  | 'rerun'
  | 'investigate'

export interface APInsight {
  id: number
  key: string
  title: string
  severity: APInsightSeverity
  body: string
  metric_value: number | null
  metric_unit: string | null
  evidence: Record<string, unknown> | null
  action_label: string | null
  action_type: APInsightActionType | null
  action_payload: Record<string, unknown> | null
  computed_at: string | null
  dismissed: boolean
}

export interface APInsightListResponse {
  items: APInsight[]
  total: number
  critical: number
  warning: number
  info: number
  decisions_analysed: number
  computed_at: string | null
}

File: app/schemas/ap_insights.py
"""Response shapes for the AI Insights surface."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class APInsight(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    key: str
    title: str
    severity: str
    body: str
    metric_value: float | None = None
    metric_unit: str | None = None
    evidence: dict[str, Any] | None = None
    action_label: str | None = None
    action_type: str | None = None
    action_payload: dict[str, Any] | None = None
    computed_at: datetime | None = None
    dismissed: bool


class APInsightListResponse(BaseModel):
    items: list[APInsight]
    total: int
    critical: int
    warning: int
    info: int
    decisions_analysed: int
    computed_at: datetime | None = None
